### ISSUE TITLE
"RuntimeError: release unlocked lock" - Autokey Does not run correctly in Fedora 30

### ISSUE THREAD

**Issue by Cardosaum** [autokey/autokey#316]
## Classification:

Bug

## Reproducibility:

Always that a script is triggered

## Version

AutoKey version: 0.95.7

Used GUI (Gtk, Qt, or both):  Both

Installed via: Fedora repository (`dnf install autokey-qt-0.95.7-2.fc30.noarch`)

Linux Distribution:

LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: Fedora
Description:    Fedora release 30 (Thirty)
Release:        30
Codename:       Thirty

## Summary

Whenever I try to run a script this error message is shown:
> RuntimeError: release unlocked lock

## Steps to Reproduce (if applicable)

- Install autokey via `dnf`
- try to execute some script

## Expected Results

- The script should run correctly (as in Debian distros.)

This problem does not occur on my Kubuntu 19.04. everything is just fine there

## Actual Results

For example, I have one script that, triggered by the phrase `&cell`, should open a browse window.
As you can see on the message below, the script is triggered correctly, however,  something happen when `service.py` is executed. 

Don't know why this happen neither how to solve this issue...

```
2019-11-01 13:30:04,580 DEBUG - iomediator - Key.SHIFT pressed
&2019-11-01 13:30:04,869 DEBUG - service - Raw key: '7', modifiers: [<Key.SHIFT: '<shift>'>], Key: &
2019-11-01 13:30:04,869 DEBUG - service - Window visible title: 'autokey : autokey-gtk — Konsole', Window class: 'konsole.konsole'
2019-11-01 13:30:04,870 DEBUG - service - Input queue at end of handle_keypress: deque(['&'], maxlen=150)
2019-11-01 13:30:05,059 DEBUG - iomediator - Key.SHIFT released
c2019-11-01 13:30:05,453 DEBUG - service - Raw key: 'c', modifiers: [], Key: c
2019-11-01 13:30:05,453 DEBUG - service - Window visible title: 'autokey : autokey-gtk — Konsole', Window class: 'konsole.konsole'
2019-11-01 13:30:05,454 DEBUG - service - Input queue at end of handle_keypress: deque(['&', 'c'], maxlen=150)
e2019-11-01 13:30:05,517 DEBUG - service - Raw key: 'e', modifiers: [], Key: e
2019-11-01 13:30:05,517 DEBUG - service - Window visible title: 'autokey : autokey-gtk — Konsole', Window class: 'konsole.konsole'
2019-11-01 13:30:05,518 DEBUG - service - Input queue at end of handle_keypress: deque(['&', 'c', 'e'], maxlen=150)
l2019-11-01 13:30:05,668 DEBUG - service - Raw key: 'l', modifiers: [], Key: l
2019-11-01 13:30:05,668 DEBUG - service - Window visible title: 'autokey : autokey-gtk — Konsole', Window class: 'konsole.konsole'
2019-11-01 13:30:05,669 DEBUG - service - Input queue at end of handle_keypress: deque(['&', 'c', 'e', 'l'], maxlen=150)
l2019-11-01 13:30:05,812 DEBUG - service - Raw key: 'l', modifiers: [], Key: l
2019-11-01 13:30:05,812 DEBUG - service - Window visible title: 'autokey : autokey-gtk — Konsole', Window class: 'konsole.konsole'
2019-11-01 13:30:05,813 INFO - service - Matched Script "runBrowser_bce_book_the_cell" having abbreviations "['&cell']" against current input
2019-11-01 13:30:05,813 DEBUG - service - Script runner executing: Script('runBrowser_bce_book_the_cell')
2019-11-01 13:30:05,814 DEBUG - service - Input queue at end of handle_keypress: deque([], maxlen=150)
2019-11-01 13:30:05,814 ERROR - service - Ignored locking error in handle_keypress
Traceback (most recent call last):
  File "/home/mcsouza/.local/lib/python3.7/site-packages/autokey/service.py", line 207, in __tryReleaseLock
    self.configManager.lock.release()
RuntimeError: release unlocked lock
2019-11-01 13:30:05,815 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2019-11-01 13:30:05,816 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2019-11-01 13:30:05,819 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2019-11-01 13:30:05,820 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]
2019-11-01 13:30:05,821 DEBUG - interface - Send special key: [<Key.BACKSPACE: '<backspace>'>]

```

## Notes

For now, I haven't found any workaround.

**Comment by josephj11** [comment]
What exactly does your script do wrong? That particular error message has been around for many years (prior to this fork!) and has never caused any functional defect, I think Thomas did something to suppress it getting issued awhile back.

**Comment by Cardosaum** [comment]
@josephj11 ,

Actually, the script isn't started (I guess).

the phrase `&cell` is indeed deleted by `<backspace>`, however, the script does not execute.

Now that you said this is a known error I ran `autokey-qt -l` on my Kubuntu and saw this same error message on logs. yet, this error does not cause any dysfunction on scripts.

I'll take a closer look on why this is not working on Fedora. probably on Monday I'll have time for this. Hope soon I update this issue.

Thanks for the attention!

**Comment by josephj11** [comment]
I see two things. The trace shows that your trigger phrase was recognized and that AutoKey emitted five backspaces to delete the trigger phrase as it is supposed to do. The included fragment of the trace does not show the ampersand being processed, but it must have been because the script `runBrowser_bce_book_the_cell` was executed.

I'm a bit surprised that the trigger phrase works with an ampersand in it. I thought that might be considered a "non-word" character. Can you run the same script using a hotkey or a trigger that doesn't include any special characters?

What does the script itself do (so you can determine if it ran successfully or not)? Is there any way it can run successfully without producing any visible output/actions?

**Comment by Cardosaum** [comment]
Sorry for the delay,

> Can you run the same script using a hotkey or a trigger that doesn't include any special characters?

I tried to do so, and no success... 
Tried with different phrase and hotkeys too, I was getting the same result.

> What does the script itself do (so you can determine if it ran successfully or not)?

This script, in particular, should open a browser in a specific web page, so, in a successfully execution, I would get a visible output.

I searched a lot, but could not find a solution. If I run these scripts manually, they all execute as expected.

I installed Fedora for the sake of curiosity, but the experience was not that good. I  opted to change to another Distro. Now, in Manjaro, Autokey is working like a charm. So, I think I'm going close this issue.
